src/task.c

```diff
@@ -1281,6 +1294,10 @@ task_handler (gpointer user_data)
       break;
     }
     case TASK_NEXT:
+      if (!task->fetch_succeeded && task->abort_recipe_on_fail) {
```

Comment:
With this change, it will proceed to get the next task but I believe your intent is to quit the current task after failing to perform the fetch operation for it.  this does not make sense.

Reply:
Yes I'd like to end on the aborted task. But I thought it should be handled by this:
            g_set_error(&task->error, RESTRAINT_ERROR, RESTRAINT_TASK_RUNNER_ABORTED, "Aborted due to fetch failure");
But it just aborts curent task not the whole recipe. So this was the mechanism to abort all the following tasks manualy with coresponding log message.

Is there a mechanizm that could abort whole recipe? I tried to skip all the following tasks in a list in the recipe. But it did not work with Beaker properly. Beaker was stuck wating for results from those skiped tasks.